splicemachine/spark/context.py

```diff
@@ -300,6 +500,45 @@ def createTable(self, schema_table_name, dataframe, keys=None, create_table_opti
         """
         if to_upper:
             dataframe = self.toUpper(dataframe)
-        # Need to convert List (keys) to scala seq
-        keys_seq = self.jvm.PythonUtils.toSeq(keys)
-        self.context.createTable(schema_table_name, dataframe._jdf.schema(), keys_seq, create_table_options)
+        self.context.createTableWithSchema(schema_table_name, df.schema, keys=keys, create_table_options=create_table_options)
+        
+    def createTableWithSchema(self, schema_table_name, schema, keys=None, create_table_options=None):
+        """
+        Creates a schema.table from a schema
+        :param schema_table_name: str The schema.table to create
+        :param schema: (StructType) The schema that describes the columns of the table
+        :param keys: List[str] The primary keys. Default None
+        :param create_table_options: str The additional table-level SQL options. Default None
+        """
+        if keys:
+            keys_seq = self.jvm.PythonUtils.toSeq(keys)
+        else:
+            keys_seq = self.jvm.PythonUtils.toSeq([])
+        self.context.createTable(
+            schema_table_name,
+            self._jstructtype(schema),
+            keys_seq,
+            create_table_options
+        )
+
+class PySpliceContext2(PySpliceContext):
```

Comment:
I think this should be called ePySpliceContext

Reply:
I’m fine with any name, though I’ve heard some people refer to the project as eNSDS and some as NSDS v2.  I’m not sure when or if one name will be settled on.  I wonder if the product team has an opinion on this, as this is customer-facing code.

Reply:
They have decided on ExtPySpliceContext

Reply:
Changed the name in new commit [4ba0818](https://github.com/splicemachine/pysplice/pull/54/commits/4ba081864cfbdc40ec28f941b632b8bf9dad8c71).